Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -2089,15 +2103,48 @@ NiftiImageIO::SetImageIOOrientationFromNIfTI(unsigned short dims)
           }();
           prefer_sform_over_qform = sform_and_qform_are_very_similar;
         }
-      } // sform is orthonormal
-    }   // sform not NIFTI_XFORM_UNKNOWN
+      }
+      else if (m_SFORM_Permissive) // sform is orthonormal
+      {
+        // Fix to deal with non-orthogonal matrixes
+        // this approach uses SVD decomposition
+        // maybe it is better to use nifti_make_orthog_mat44
+        // to be consistent with other software?
```

Comment:
Unfortunately the block below doesn't produce orthogonal matrix. The test image _NonOrthoSform.nii.gz_ ends up with
```
  Direction: 
0.694449 0.264737 0.469327
-0.625039 -0.670148 -0.125412
0.356465 -0.693409 0.874073
```
The matrix is not orthogonal. I have tested the alternative you mentioned above and it works very good, `nifti_make_orthog_mat44` generates orthogonal matrices.

```cpp
      else if (m_SFORM_Permissive)
      {
        itkWarningMacro(<< this->GetFileName() << " has non-orthogonal sform, corrected");
        this->m_SFORM_Corrected = true;
        return nifti_make_orthog_mat44(this->m_NiftiImage->sto_xyz.m[0][0],
                                       this->m_NiftiImage->sto_xyz.m[0][1],
                                       this->m_NiftiImage->sto_xyz.m[0][2],
                                       this->m_NiftiImage->sto_xyz.m[1][0],
                                       this->m_NiftiImage->sto_xyz.m[1][1],
                                       this->m_NiftiImage->sto_xyz.m[1][2],
                                       this->m_NiftiImage->sto_xyz.m[2][0],
                                       this->m_NiftiImage->sto_xyz.m[2][1],
                                       this->m_NiftiImage->sto_xyz.m[2][2]);
      }
```

Reply:
Updated the above post (to trigger notification).